packages/utils/src/store/Store.ts

```diff
@@ -47,10 +51,22 @@ export class Store<State> {
    *
    * @param newState The new state to set for the store.
    */
-  public update(newState: State) {
-    if (this.state !== newState) {
-      this.state = newState;
-      this.listeners.forEach((l) => l(newState));
+  public setState(newState: State) {
```

Comment:
Having both `set` and `setState` isn't ideal, but I can't think of a better name. `replaceState` describes what's happening, but I'm not sure I like it.

We could also rename `set` to `setField`, but since it's commonly used, it's better to have it short.